modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/SqlRowHandler.java

```diff
@@ -440,4 +424,54 @@ RowSchema rowSchema() {
             }
         }
     }
+
+    private static class RowBuilderImpl implements RowBuilder<RowWrapper> {
+
+        private final int schemaLen;
+
+        private final RowSchema rowSchema;
+
+        Object[] data;
+
+        int fieldIdx;
+
+        RowBuilderImpl(RowSchema rowSchema) {
+            this.rowSchema = rowSchema;
+            this.schemaLen = rowSchema.fields().size();
+            fieldIdx = 0;
+        }
+
+        /** {@inheritDoc} */
+        @Override
+        public RowBuilder<RowWrapper> newRow() {
+            data = new Object[schemaLen];
+            fieldIdx = 0;
+            return this;
+        }
+
+        /** {@inheritDoc} */
+        @Override
+        public RowBuilder<RowWrapper> addField(Object value) {
+            checkState();
+
+            data[fieldIdx++] = value;
+            return this;
+        }
+
+        /** {@inheritDoc} */
+        @Override
+        public RowWrapper build() {
+            checkState();
+
+            Object[] row = data;
+            data = null;
+            fieldIdx = 0;
+
+            return new ObjectsArrayRowWrapper(rowSchema, row);
+        }
+
+        private void checkState() {
+            assert data != null : "Row has not been initialised";
```

Comment:
```suggestion
            if (data == null) {
                new IllegalStateException("Row has not been initialised");
            }
```

Reply:
Otherwise, ObjectsArrayRowWrapper could be created with `null` array, which lead to NPE in a random place, and is harder to investigate if assertions are disabled.

Reply:
Fixed.